### ISSUE TITLE
KVM:`Requested operation is not valid: domain is already running`

### ISSUE THREAD

**Issue by scottkurz** [kubernetes/minikube#11403]
**Steps to reproduce the issue:**

1.  On a Centos 8.3 VM (provisioned via private cloud) .... setup kvm/libvirt as described here: https://phoenixnap.com/kb/install-minikube-on-centos (in section 2.)
2.  Install **minikube version: v1.20.0** and configure: `minikube config set driver kvm2`  # plus set cpus, memory
3. minikube start

**Full output of `minikube logs` command:**

Attached here as: [lastStart.txt](https://github.com/kubernetes/minikube/files/6474002/lastStart.txt)

I kind of cheated in that I pasted here not the original failure but a recreate after `virsh destroy minikube` cleanup after the initial failure, but I think I'm recreating the same problem (after the cleanup).

I think the key part is here:

<details>

```
I0513 09:51:46.235129    2576 out.go:170] 🔄  Restarting existing kvm2 VM for "minikube" ...
I0513 09:51:46.235163    2576 main.go:128] libmachine: (minikube) Calling .Start
I0513 09:51:46.235370    2576 main.go:128] libmachine: (minikube) Ensuring networks are active...
I0513 09:51:46.252118    2576 main.go:128] libmachine: (minikube) Ensuring network default is active
I0513 09:51:46.252554    2576 main.go:128] libmachine: (minikube) Ensuring network mk-minikube is active
I0513 09:51:46.252919    2576 main.go:128] libmachine: (minikube) Getting domain xml...
I0513 09:51:46.269085    2576 main.go:128] libmachine: (minikube) Creating domain...
I0513 09:51:46.759011    2576 main.go:128] libmachine: (minikube) Waiting to get IP...
I0513 09:51:46.760069    2576 main.go:128] libmachine: (minikube) DBG | domain minikube has defined MAC address 52:54:00:67:ae:81 in network mk-minikube
I0513 09:51:46.760682    2576 main.go:128] libmachine: (minikube) DBG | unable to find current IP address of domain minikube in network mk-minikube
I0513 09:51:46.760767    2576 main.go:128] libmachine: (minikube) DBG | I0513 09:51:46.760652    2611 retry.go:31] will retry after 263.082536ms: waiting for machine to come up
I0513 09:51:47.025437    2576 main.go:128] libmachine: (minikube) DBG | domain minikube has defined MAC address 52:54:00:67:ae:81 in network mk-minikube
I0513 09:51:47.025942    2576 main.go:128] libmachine: (minikube) DBG | unable to find current IP address of domain minikube in network mk-minikube
I0513 09:51:47.025965    2576 main.go:128] libmachine: (minikube) DBG | I0513 09:51:47.025908    2611 retry.go:31] will retry after 381.329545ms: waiting for machine to come up
I0513 09:51:47.408478    2576 main.go:128] libmachine: (minikube) DBG | domain minikube has defined MAC address 52:54:00:67:ae:81 in network mk-minikube
I0513 09:51:47.408999    2576 main.go:128] libmachine: (minikube) DBG | unable to find current IP address of domain minikube in network mk-minikube
I0513 09:51:47.409036    2576 main.go:128] libmachine: (minikube) DBG | I0513 09:51:47.408938    2611 retry.go:31] will retry after 422.765636ms: waiting for machine to come up
I0513 09:51:47.833772    2576 main.go:128] libmachine: (minikube) DBG | domain minikube has defined MAC address 52:54:00:67:ae:81 in network mk-minikube
....
I0513 09:52:27.553736    2576 main.go:128] libmachine: (minikube) DBG | unable to find current IP address of domain minikube in network mk-minikube
I0513 09:52:27.553761    2576 main.go:128] libmachine: (minikube) DBG | I0513 09:52:27.553677    2611 retry.go:31] will retry after 9.953714808s: waiting for machine to come up
I0513 09:52:37.509293    2576 main.go:128] libmachine: (minikube) DBG | domain minikube has defined MAC address 52:54:00:67:ae:81 in network mk-minikube
I0513 09:52:37.509690    2576 main.go:128] libmachine: (minikube) DBG | unable to find current IP address of domain minikube in network mk-minikube
I0513 09:52:37.509735    2576 main.go:128] libmachine: (minikube) DBG | I0513 09:52:37.509669    2611 retry.go:31] will retry after 15.120437328s: waiting for machine to come up
I0513 09:52:52.632670    2576 main.go:128] libmachine: (minikube) DBG | domain minikube has defined MAC address 52:54:00:67:ae:81 in network mk-minikube
I0513 09:52:52.633018    2576 main.go:128] libmachine: (minikube) DBG | unable to find current IP address of domain minikube in network mk-minikube
I0513 09:52:52.633665    2576 fix.go:57] fixHost completed within 1m6.449108061s
I0513 09:52:52.633685    2576 start.go:80] releasing machines lock for "minikube", held for 1m6.44914207s
W0513 09:52:52.633729    2576 start.go:505] error starting host: driver start: IP not available after waiting: machine minikube didn't return IP after 1 minute
W0513 09:52:52.633942    2576 out.go:235] 🤦  StartHost failed, but will try again: driver start: IP not available after waiting: machine minikube didn't return IP after 1 minute
```

</details>

It also surfaces the error messages:

```
W0513 09:52:57.724005    2576 out.go:235] 😿  Failed to start kvm2 VM. Running "minikube delete" may fix it: driver start: error creating VM: virError(Code=55, Domain=10, Message='Requested operation is not valid: domain is already running')
I0513 09:52:57.727508    2576 out.go:170]
W0513 09:52:57.727632    2576 out.go:235] ❌  Exiting due to GUEST_PROVISION: Failed to start host: driver start: error creating VM: virError(Code=55, Domain=10, Message='Requested operation is not valid: domain is already running')
```

But I think the "domain is already running" is a consequence of the root cause here.

**Full output of failed command:**
<details>

==> Audit <==
|---------|-----------------|----------|----------|---------|-------------------------------|-------------------------------|
| Command |      Args       | Profile  |   User   | Version |          Start Time           |           End Time            |
|---------|-----------------|----------|----------|---------|-------------------------------|-------------------------------|
| config  | set memory 4096 | minikube | ibmadmin | v1.20.0 | Thu, 13 May 2021 09:24:15 PDT | Thu, 13 May 2021 09:24:15 PDT |
| config  | set cpus 4      | minikube | ibmadmin | v1.20.0 | Thu, 13 May 2021 09:24:19 PDT | Thu, 13 May 2021 09:24:19 PDT |
| config  | set driver kvm2 | minikube | ibmadmin | v1.20.0 | Thu, 13 May 2021 09:24:23 PDT | Thu, 13 May 2021 09:24:23 PDT |
|---------|-----------------|----------|----------|---------|-------------------------------|-------------------------------|

==> Last Start <==
(attached above).

</details>

**Other observations** 

1. Seems like there's some overlap with:   
https://github.com/kubernetes/minikube/issues/9381
https://github.com/kubernetes/minikube/issues/10102

2.  From the fact that there's a whole troubleshooting section here:   https://minikube.sigs.k8s.io/docs/drivers/kvm2/#troubleshooting-kvmlibvirt-networks  I take it this kind of thing has been seen before.   But, not sure what to do, and from scanning the other GitHub issues it seemed the maintainers were offering to help, so here you go (thank you).  

The one thing I thought to do from that section was run:

```
$ sudo virsh net-list --all

 Name          State    Autostart   Persistent
------------------------------------------------
 default       active   yes         yes
 mk-minikube   active   yes         yes
```

**Comment by medyagh** [comment]
seems like we are not handling the casae that `Requested operation is not valid: domain is already running'

@prezha is our KVM expert, he recently improved the KVM handling the networks, any suggestions @prezha ?

**Comment by scottkurz** [comment]
Thx @medyagh, no,  that didn't help.   I might not have made this clear but this happens on a brand new VM on a fresh installation of minikube...nothing cached in user dir or anywhere.

TBH,  I don't really know what's going on but I think the "..domain is already running" is a secondary symptom.  (I thought it might help others search/match this issue.)

The way I interpret the error message...1)  it tries to create the 'minikube'     2) it succeeds but it fails to establish connectivity   3) it retries to create 'minikube' and this time fails since it already exists.

(E.g. this excerpt).
```
W0513 13:54:23.292041    6750 start.go:505] error starting host: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute
🤦  StartHost failed, but will try again: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute
I0513 13:54:23.517822    6750 start.go:520] Will try again in 5 seconds ..
```

**Comment by medyagh** [comment]
is this a VM in the cloud ? does your cloud allow nested virtualization ?
https://cloud.google.com/compute/docs/instances/enable-nested-virtualization-vm-instances 

if not, have u tried the Docker driver on the VM ?

**Comment by medyagh** [comment]
would u also please share the output of
```
$ virt-host-validate
```

**Comment by prezha** [comment]
@medyagh thanks for such kind words :) i'll have a look

@scottkurz thank you for opening this issue and providing details so far!

the confusing part in the "lastStart.txt" is: `Skipping create...Using existing machine configuration` and `Restarting existing kvm2 VM for "minikube"` - if that was the result of running `minikube start` in a "clean-ed" environment (ie, no minikube clusters at all), and therefore, the `domain is already running` is then even more confusing

if i can kindly ask you to run the following commands-sequence and share outputs:
- `minikube config view`
- `minikube delete --all --purge`
- `virsh list --all`
- `sudo virsh list --all`
- `echo $LIBVIRT_DEFAULT_URI`
- `minikube start --driver=kvm2 --alsologtostderr -v=9` and attach the full log output here please

thanks!

**Comment by scottkurz** [comment]
Hmm... now I'm 90% sure the issue is my VM doesn't support nested virtualization.   

I did: 
> /usr/libexec/qemu-kvm   -hda debian_squeeze_amd64_standard.qcow2 -m 512

and got what looks like a CPU dump, and see a similar dump in:   **/var/log/libvirt/qemu/minikube.log**  .   And I also tried a couple `virt-install` commands to create VMs, stuck in paused state with log messages that seemed consistent with nested virt. not supported.

So my apologies.... I would have assumed this was the problem except that I had just provisioned one where nested virtualization worked.   I'm still getting my head around what the difference is, but it's an internal cloud, so the details aren't probably generally helpful anyway.

But thanks for your time.  Sorry the problem was all on my end.   I'll close assuming there's nothing helpful to be done here though if you're interested I'd still gather logs if you asked.
